Add options to hide sea creature catch and fishing treasure messages

LeaPhant opened the pull request:
_bear with me I haven't written a single line of Kotlin in my life_

LeaPhant committed:
Add spam hider for sc catches and fishing treasures

LeaPhant committed:
Separate toggles for good and great fishing treasures

DJtheRedstoner commented on src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpamHider.kt:
```diff
@@ -69,6 +69,43 @@ class SpamHider {
     private var abilityUses = 0
     private var lastAbilityUsed = ""
 
+    private var SEA_CREATURES = setOf<String>(
```
This should be a val (means its final/constant), not var and the explicit generic is not required.

DJtheRedstoner commented on src/main/kotlin/skytils/skytilsmod/core/Config.kt:
```diff
@@ -2207,6 +2207,46 @@ class Config : Vigilant(File("./config/skytils/config.toml"), "Skytils", sorting
+    @Property(
+        type = PropertyType.SELECTOR,
+        name = "Legendary Sea Creature Catch Hider",
+        description = "Removes legendary sea creature catch messages as well.",
```
"as well" sounds a bit informal to me.

DJtheRedstoner commented on src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpamHider.kt:
```diff
@@ -69,6 +69,43 @@ class SpamHider {
+        "You have awoken the Deep Sea Protector, prepare for a battle!",
+    )
+
+    private var LEGENDARY_SEA_CREATURES = setOf<String>(
```
Same as above.

LeaPhant committed:
Change phrasing of description

Resolves https://github.com/Skytils/SkytilsMod/pull/31#discussion_r645019347

LeaPhant committed:
Use val instead of var for constant

Resolves https://github.com/Skytils/SkytilsMod/pull/31#discussion_r645017997 and https://github.com/Skytils/SkytilsMod/pull/31#discussion_r645019397

Sychic commented on src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpamHider.kt:
```diff
@@ -380,6 +417,42 @@ class SpamHider {
+                }
+            }
+
+            // Good catch
```
Could be changed to use formatted but doesn't really matter.

My-Name-Is-Jeff replied:
The only issue with using unformatted + contains is that it can trigger on player chat, even though it's unlikely

LeaPhant replied:
Since contains checks against the set in this case it should only trigger if the full chat message is exactly that phrase which a player wouldn't be able to send.

Sychic commented on src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpamHider.kt:
```diff
@@ -380,6 +417,42 @@ class SpamHider {
+                }
+            }
+
+            // Great catch
```
This one too.

My-Name-Is-Jeff replied:
Should be fine because it is impossible for another player to send a message that starts with this
